python/hsml/utils/tensor_schema.py

```diff
@@ -0,0 +1,72 @@
+#
+#   Copyright 2021 Logical Clocks AB
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+from hsml.utils.tensor import Tensor
+import numpy
+
+
+class TensorSchema:
+    """Metadata object representing a tensor schema for a model."""
+
+    def __init__(self, tensor_obj=None):
+
+        if isinstance(tensor_obj, list):
+            self.tensors = self._convert_list_to_schema(tensor_obj)
+        elif isinstance(tensor_obj, numpy.ndarray):
+            self.tensors = self._convert_tensor_to_schema(tensor_obj)
+        else:
+            raise TypeError("{} is not supported in a schema.".format(type(tensor_obj)))
```

Comment:
```suggestion
            raise TypeError("{} is not supported in a tensor schema.".format(type(tensor_obj)))
```
Since columnar and tensor schemas don't support the same types, I think it's better if we specify the schema type with which the current object is not supported.